src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,192 @@
+        try {
+            if (taskList.size() == 0) {
+                reloadList();
+            }
+            Todo todo = new Todo(taskName);
+            taskList.add(todo);
+            Storage.addToList(TaskType.TODO, 0, taskName, "");
+            String result = "Got it. I've added the task:\n" + todo.toString() + "\n";
+            result += "Now you have " + taskList.size() + " tasks in the list.";
+            return Ui.userMessage(result);
+        } catch (IOException e) {
+            throw new IOException("File access failed");
+        }
+    }
+
+    /**
+     * Creates a task and adds it to the task list: deadline or event
+     *
+     * @param type     task type
+     * @param taskName task name
+     * @param time     time of the task
+     * @return messages of the added task or errors occurred
+     * @throws IOException if error occurs in file accessing
+     */
+    public static String add(TaskType type, String taskName, String time) throws IOException {
+        try {
+            if (taskList.size() == 0) {
+                reloadList();
+            }
+            String result = "Got it. I've added the task:\n";
+            if (type == TaskType.DEADLINE) {
+                Deadline deadline = new Deadline(taskName, time);
+                taskList.add(deadline);
+                Storage.addToList(TaskType.DEADLINE, 0, taskName, time);
+                result += deadline.toString();
+            } else {
+                Event event = new Event(taskName, time);
+                taskList.add(event);
+                Storage.addToList(TaskType.EVENT, 0, taskName, time);
+                result += event.toString();
+            }
+            result += "\nNow you have " + taskList.size() + " tasks in the list.";
+            return Ui.userMessage(result);
+        } catch (IOException e) {
+            throw new IOException("File access failed");
+        }
+
+    }
+
+    private static void reloadList() throws IOException {
+        taskList = Storage.readList();
+    }
+
+    /**
+     * Finds tasks that contain a given keyword
+     *
+     * @param word keyword of the task
+     * @return a string representation of the tasks found
+     * @throws IOException               if error occurs in file accessing
+     * @throws InvalidParameterException if word is blank
+     */
+    public static String findTask(String word) throws IOException, InvalidParameterException {
+        if (word.isBlank()) {
+            throw new InvalidParameterException("Invalid input");
+        }
+        int n = taskList.size();
+        if (n == 0) { // empty or uninitialized
+            reloadList();
+            n = taskList.size();
+            if (n == 0) {
+                return Ui.listError();
+            }
+        }
+        StringBuilder sb = new StringBuilder();
+        sb.append("Here are the matching tasks in your list:");
+        int count = 1;
+        boolean found = false;
```

Comment:
_found_ sounds alright but I don't know if it is required for it to be named as a question. 🤔